cmd/kubelet/app/server.go

```diff
@@ -844,35 +849,44 @@ func buildKubeletClientConfig(s *options.KubeletServer, nodeName types.NodeName)
 }
 
 // updateDialer instruments a restconfig with a dial. the returned function allows forcefully closing all active connections.
-func updateDialer(clientConfig *restclient.Config) (func(), error) {
-	if clientConfig.Transport != nil || clientConfig.Dial != nil {
-		return nil, fmt.Errorf("there is already a transport or dialer configured")
+func updateDialer(clientConfigs *restclient.Config) ([]func(), error) {
+	returnFuncs := make([]func(), len(clientConfigs.GetAllConfigs()))
+	for i, clientConfig := range clientConfigs.GetAllConfigs() {
+		if clientConfig.Transport != nil || clientConfig.Dial != nil {
+			return nil, fmt.Errorf("there is already a transport or dialer configured")
+		}
+		d := connrotation.NewDialer((&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext)
```

Comment:
If we have more than 10 clients,  is that possible to wait 300 seconds to get timeout?

Reply:
I mean we can make dialers asynchronously.

Reply:
That is the plan. I forgot to add todo note here.

Reply:
The actual dialing is not happening here so here does not need change.